pkg/cmd/claude/remote.go

```diff
@@ -0,0 +1,296 @@
+			fmt.Fprintf(stdout, "Session ID: %s\n", sessionID)
+			fmt.Fprintf(stdout, "The session will continue running in the background.\n")
+			return nil
+		case <-ticker.C:
+			getReq := &agentv1.GetRemoteSessionRequest{
+				SessionId:      sessionID,
+				OrganizationId: orgID,
+			}
+			getResp, err := client.GetRemoteSession(ctx, api.WithAuthentication(connect.NewRequest(getReq), token))
+			if err != nil {
+				var connectErr *connect.Error
+				if errors.As(err, &connectErr) && connectErr.Code() == connect.CodeNotFound {
+					continue // Continue waiting
+				}
+				return fmt.Errorf("failed to get remote session status: %w", err)
+			}
+
+			if getResp.Msg.StartedAt == nil {
+				continue
+			}
+			sessionStartTime := getResp.Msg.StartedAt.AsTime()
+			if !sessionStartTime.After(invocationTime) {
+				continue
+			}
+
+			if !sessionStarted {
+				sessionStarted = true
+				fmt.Fprintf(stdout, "\n✓ Remote session started!\n")
+				fmt.Fprintf(stdout, "Session ID: %s\n", sessionID)
+			}
+
+			if getResp.Msg.CompletedAt != nil {
+				exitCode := 0
+				if getResp.Msg.ExitCode != nil {
+					exitCode = int(*getResp.Msg.ExitCode)
+				}
+
+				var err error
+				if exitCode != 0 {
+					fmt.Fprintf(stdout, "\n✗ Remote session exited with code %d\n", exitCode)
+					fmt.Fprintf(stdout, "Session ID: %s\n", sessionID)
+					//fmt.Fprintf(stderr, "View session logs: https://depot.dev/orgs/%s/claude/sessions/%s\n", orgID, sessionID)
+					if getResp.Msg.ErrorMessage != nil && *getResp.Msg.ErrorMessage != "" {
+						fmt.Fprintf(stdout, "Error: %s\n", *getResp.Msg.ErrorMessage)
+					}
+					err = fmt.Errorf("remote session exited with code %d", exitCode)
+				} else {
+					fmt.Fprintf(stdout, "\n✓ Remote session exited successfully (exit code 0)\n")
+					fmt.Fprintf(stdout, "Session ID: %s\n", sessionID)
+					//fmt.Fprintf(stdout, "View full session: https://depot.dev/orgs/%s/claude/sessions/%s\n", orgID, sessionID)
+				}
+
+				if getResp.Msg.DurationSeconds != nil {
+					fmt.Fprintf(stdout, "Duration: %.2f seconds\n", *getResp.Msg.DurationSeconds)
+				}
+				return err
+
+			}
+
+			if !sessionRunning {
+				req := &agentv1.DownloadClaudeSessionRequest{
+					SessionId: sessionID,
+				}
+				if orgID != "" {
+					req.OrganizationId = &orgID
+				}
+
+				_, err := client.DownloadClaudeSession(ctx, api.WithAuthentication(connect.NewRequest(req), token))
+				if err == nil {
+					sessionRunning = true
+					fmt.Fprintf(stdout, "Remote session is running. Output will be saved and can be viewed at the URL above.\n")
+					fmt.Fprintf(stdout, "You can resume this session later with: depot claude --resume %s\n", sessionID)
+					fmt.Fprintf(stdout, "\nWaiting for session to complete...\n")
+				}
+			}
+		}
+	}
+}
+
+func checkRequiredClaudeSecrets(ctx context.Context, client agentv1connect.ClaudeServiceClient, token, orgID string, stderr io.Writer) error {
+	req := &agentv1.ListSecretsRequest{}
```

Comment:
This will only work if the person is an org owner I believe. Because this is server data, perhaps it could be checked serverside?